dbms/src/Common/UniThreadPool.cpp

```diff
@@ -369,9 +368,12 @@ GlobalThreadPool & GlobalThreadPool::instance()
 
 GlobalThreadPool::~GlobalThreadPool() noexcept
 {
-    // We must make sure IOThread is released before GlobalThreadPool runs
+    // We must make sure sub thread pools are released before GlobalThreadPool runs
     // `finalize`. Or the threads in GlobalThreadPool never ends.
-    IOThreadPool::shutdown();
+
+    // FIXME: Make this less coupled.
+    GeneralIOThreadPool::shutdown();
+    S3IOThreadPool::shutdown();
```

Comment:
We are not referencing RNPagePreparerThreadPool here because I feel it not a right place. cc @JaySon-Huang